Runtime/Data/Type/Transformation/Vector2MagnitudeSetter.cs

```diff
@@ -0,0 +1,48 @@
+﻿namespace Zinnia.Data.Type.Transformation
+{
+    using UnityEngine;
+    using UnityEngine.Events;
+    using System;
+    using Malimbe.PropertySerializationAttribute;
+    using Malimbe.XmlDocumentationAttribute;
+
+    /// <summary>
+    /// Transforms a <see cref="Vector2"/> by changing its magnitude.
+    /// </summary>
+    public class Vector2MagnitudeSetter : Transformer<Vector2, Vector2, Vector2MagnitudeSetter.UnityEvent>
+    {
+        /// <summary>
+        /// Defines the event with the <see cref="Vector2"/> with the changed magnitude.
+        /// </summary>
+        [Serializable]
+        public class UnityEvent : UnityEvent<Vector2>
+        {
+        }
+
+        /// <summary>
+        /// The magnitude to use when transforming values.
+        /// </summary>
+        [Serialized]
+        [field: DocumentedByXml]
+        public float Magnitude { get; set; } = 1f;
+
+        /// <summary>
+        /// Sets the magnitude to use.
+        /// </summary>
+        /// <param name="magnitudeSource">The source of the magnitude to use when transforming values.</param>
+        public virtual void SetMagnitude(Vector2 magnitudeSource)
+        {
+            Magnitude = magnitudeSource.magnitude;
```

Comment:
This isn't needed, why not just set the `Magnitude` property directly?

Reply:
this setter is dup from Vector3MagnitudeSetter
whatever is there i have it here

Reply:
Ah right I see now, this is based on a really old script I think @bddckr added to do the body representation stuff.

Ok, so do you need right now a `Vector2MagnitudeSetter`? Because I'd say, pull that component out of this PR completely because I think these MagnitudeSetters need a rethink and refactor as they don't make sense.

Reply:
this is the closest thing to "extract the .normalized" that you mentioned earlier
so i added it in just for completeness 
i can pull this out